test/e2e/dns_autoscaling.go

```diff
@@ -207,14 +226,16 @@ func deleteDNSAutoscalerPod(c clientset.Interface) error {
 	return nil
 }
 
-func waitForDNSReplicasSatisfied(c clientset.Interface, expected int, timeout time.Duration) (err error) {
+func waitForDNSReplicasSatisfied(c clientset.Interface, getExpected getExpectReplicasFunc, timeout time.Duration) (err error) {
 	var current int
+	expected := getExpected(c)
 	framework.Logf("Waiting up to %v for kube-dns reach %v replicas", timeout, expected)
 	condition := func() (bool, error) {
 		current, err = getDNSReplicas(c)
 		if err != nil {
 			return false, err
 		}
+		expected = getExpected(c)
 		if current != expected {
```

Comment:
This test will not timeout because it is not actually counting the running pods. Instead it just gets the replicas count from kube-dns deployment.

And because it is using the `GetReadySchedulableNodesOrDie()` function from framework, I think it does print out warning if a node is not ready, like below:
```
Nov 10 00:36:26.951: INFO: Condition Ready of node jenkins-e2e-minion-group-x6w1 is false instead of true.
```